src/test/java/ca/islandora/syn/valves/SynFilterTest.java

```diff
@@ -0,0 +1,674 @@
+    private ArgumentCaptor<HttpServletResponse> responseCaptor;
+
+    @Before
+    public void setUp() throws Exception {
+        settings = temporaryFolder.newFile();
+        createSettings(settings);
+
+        when(config.getInitParameter("settings-path")).thenReturn(settings.getAbsolutePath());
+
+        synFilter = createFilter();
+
+        when(request.getScheme()).thenReturn("http");
+        when(request.getServerPort()).thenReturn(80);
+
+        offset = ZoneId.systemDefault().getRules().getOffset(Instant.now());
+
+        requestCaptor = ArgumentCaptor.forClass(HttpServletRequest.class);
+        responseCaptor = ArgumentCaptor.forClass(HttpServletResponse.class);
+    }
+
+    private SynFilter createFilter() throws ServletException {
+        final SynFilter synFilter = new SynFilter();
+        synFilter.init(config);
+        return synFilter;
+    }
+
+    @Test(expected = ServletException.class)
+    public void missingInitParameter() throws Exception {
+        when(config.getInitParameter("settings-path")).thenReturn(null);
+        synFilter = createFilter();
+    }
+
+    @Test(expected = ServletException.class)
+    public void absoluteSettingsDoesNotExist() throws Exception {
+        when(config.getInitParameter("settings-path")).thenReturn("/tmp/fileIsFake");
+        synFilter = createFilter();
+    }
+
+    @Test(expected = ServletException.class)
+    public void relativeSettingsDoesNotExist() throws Exception {
+        when(config.getInitParameter("settings-path")).thenReturn("fileIsFake");
+        synFilter = createFilter();
+    }
+
+    // For some reason files on the classpath are not found???
+    @Ignore
+    @Test
+    public void loadRelativeSettings() throws Exception {
+        when(config.getInitParameter("settings-path")).thenReturn("exampleSettings.yaml");
+        synFilter = createFilter();
+        final String host = "http://test.com";
+        final String username = "bob";
+        final List<String> finalRoles = Arrays.asList("islandora", host);
+
+        final String token = "Bearer " + JWT
+                .create()
+                .withClaim("webid", 1)
+                .withClaim("sub", username)
+                .withClaim("iss", host)
+                .withArrayClaim("roles", new String[] {})
+                .withIssuedAt(Date.from(LocalDateTime.now().toInstant(offset)))
+                .withExpiresAt(Date.from(LocalDateTime.now().plusHours(2).toInstant(offset)))
+                .sign(Algorithm.HMAC256("bobPassword"));
+
+        when(request.getHeader("Authorization")).thenReturn(token);
+
+        synFilter.doFilter(request, response, chain);
+
+        verify(chain).doFilter(requestCaptor.capture(), responseCaptor.capture());
+
+        assertEquals(username, requestCaptor.getValue().getUserPrincipal().getName());
+
+        for (final String role : finalRoles) {
+            assertTrue(requestCaptor.getValue().isUserInRole(role));
```

Comment:
Unless order is important, prefer `finalRoles.forEach(role -> assertTrue(requestCaptor.getValue().isUserInRole(role)))`